test/unit/specs/store/search-store.spec.js

```diff
@@ -183,55 +229,66 @@ describe('Search Store', () => {
           Promise.resolve({ data: searchData })
         ),
         getMediaDetail: jest.fn(() =>
-          Promise.resolve({ data: imageDetailData })
+          Promise.resolve({ data: audioDetailData })
         ),
       }
+      services = { [AUDIO]: audioServiceMock, [IMAGE]: imageServiceMock }
       commit = jest.fn()
       dispatch = jest.fn()
       state = {
+        audios: [{ id: 'foo' }, { id: 'bar' }, { id: 'zeta' }],
         images: [{ id: 'foo' }, { id: 'bar' }, { id: 'zeta' }],
         query: { q: 'foo query' },
       }
       rootState = { user: { usageSessionId: 'foo' } }
       store.rootState = rootState
     })
 
-    it('FETCH_MEDIA on success', (done) => {
+    it('FETCH_MEDIA throws an error on unknown media type', async () => {
+      const action = store.actions(services)[FETCH_MEDIA]
+      const params = {
+        mediaType: 'unknown',
+        page: 1,
+      }
+      try {
+        await action({ commit, dispatch, state }, params)
+      } catch (error) {
+        expect(error.message).toEqual(
+          `Cannot fetch unknown media type "${params.mediaType}"`
+        )
+      }
```

Comment:
This test could unexpectedly pass if the action never throws. I think to write this it should be something like `expect(action({ ... }).rejects(...)`: https://jestjs.io/docs/expect#rejects

That way it will fail if the action doesn't throw.

Reply:
Thank you for adding the link to docs, @sarayourfriend . I tried using `throws`, but I was using it incorrectly. And big thank you for marking all the places where the change was needed. Greatly simplified the task :)